src/modules/vtol_att_control/standard.cpp

```diff
@@ -337,78 +337,78 @@ void Standard::update_fw_state()
-				_actuators_fw_in->control[actuator_controls_s::INDEX_PITCH];
+	switch (_vtol_schedule.flight_mode) {
+	case vtol_mode::MC_MODE:
 
-			_actuators_out_1->control[actuator_controls_s::INDEX_YAW] = 0.0f;
-			_actuators_out_1->control[actuator_controls_s::INDEX_AIRBRAKES] = 0.0f;
-		}
-	}
+		// MC out = MC in
+		mc_out[actuator_controls_s::INDEX_ROLL]         = mc_in[actuator_controls_s::INDEX_ROLL];
+		mc_out[actuator_controls_s::INDEX_PITCH]        = mc_in[actuator_controls_s::INDEX_PITCH];
+		mc_out[actuator_controls_s::INDEX_YAW]          = mc_in[actuator_controls_s::INDEX_YAW];
+		mc_out[actuator_controls_s::INDEX_THROTTLE]     = mc_in[actuator_controls_s::INDEX_THROTTLE];
+		mc_out[actuator_controls_s::INDEX_LANDING_GEAR] = mc_in[actuator_controls_s::INDEX_LANDING_GEAR];
+
+		// FW out = 0, other than roll and pitch depending on elevon lock
+		fw_out[actuator_controls_s::INDEX_ROLL]         = elevon_lock ? 0 : fw_in[actuator_controls_s::INDEX_ROLL];
+		fw_out[actuator_controls_s::INDEX_PITCH]        = elevon_lock ? 0 : fw_in[actuator_controls_s::INDEX_PITCH];
+		fw_out[actuator_controls_s::INDEX_YAW]          = 0;
+		fw_out[actuator_controls_s::INDEX_THROTTLE]     = _pusher_throttle;
+		fw_out[actuator_controls_s::INDEX_FLAPS]        = 0;
+		fw_out[actuator_controls_s::INDEX_AIRBRAKES]    = 0;
 
-	// set the fixed wing throttle control
-	if (_vtol_schedule.flight_mode == vtol_mode::FW_MODE) {
+		break;
 
-		// take the throttle value commanded by the fw controller
-		_actuators_out_1->control[actuator_controls_s::INDEX_THROTTLE] =
-			_actuators_fw_in->control[actuator_controls_s::INDEX_THROTTLE];
+	case vtol_mode::TRANSITION_TO_FW:
 
-	} else {
-		// otherwise we may be ramping up the throttle during the transition to fw mode
-		_actuators_out_1->control[actuator_controls_s::INDEX_THROTTLE] = _pusher_throttle;
-	}
+	// FALLTHROUGH
+	case vtol_mode::TRANSITION_TO_MC:
+		// MC out = MC in (weighted)
+		mc_out[actuator_controls_s::INDEX_ROLL]         = mc_in[actuator_controls_s::INDEX_ROLL]     * _mc_roll_weight;
+		mc_out[actuator_controls_s::INDEX_PITCH]        = mc_in[actuator_controls_s::INDEX_PITCH]    * _mc_pitch_weight;
+		mc_out[actuator_controls_s::INDEX_YAW]          = mc_in[actuator_controls_s::INDEX_YAW]      * _mc_yaw_weight;
+		mc_out[actuator_controls_s::INDEX_THROTTLE]     = mc_in[actuator_controls_s::INDEX_THROTTLE] * _mc_throttle_weight;
+		mc_out[actuator_controls_s::INDEX_LANDING_GEAR] = 0;
+
+		// FW out = FW in, with VTOL transition controlling throttle and airbrakes
+		fw_out[actuator_controls_s::INDEX_ROLL]         = fw_in[actuator_controls_s::INDEX_ROLL];
+		fw_out[actuator_controls_s::INDEX_PITCH]        = fw_in[actuator_controls_s::INDEX_PITCH];
+		fw_out[actuator_controls_s::INDEX_YAW]          = fw_in[actuator_controls_s::INDEX_YAW];
+		fw_out[actuator_controls_s::INDEX_THROTTLE]     = _pusher_throttle;
+		fw_out[actuator_controls_s::INDEX_FLAPS]        = 0;
```

Comment:
We may as well route through the flaps as well, no? Applying them during the transition would be quite reasonable actually.

Reply:
Will do.

Reply:
Done.